modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -278,6 +278,8 @@ public DistributionZoneManager(
     @Override
     public CompletableFuture<Void> startAsync(ComponentContext componentContext) {
         return inBusyLockAsync(busyLock, () -> {
+            partitionDistributionResetTimeoutConfiguration.init();
```

Comment:
There is not enough test, maybe something else is broken after catalog compaction and restarting the node.

Reply:
> Also, is there a test that checks that the fix fixes the problem?

This fix is about the wrong `assert`, but when I fixed it I found out the issue with ItHighAvailablePartitionsRecoveryTest#testHaRecoveryOnZoneTimersRestoreAfterNodeRestart, because assertion starts to work. So, this test fixed here also.

> There is not enough test, maybe something else is broken after catalog compaction and restarting the node.
> Or will it be fixed in another ticket?

The root cause of compaction issues is the https://issues.apache.org/jira/browse/IGNITE-24057, I thought that we already on the same page with it. So, general fix and tests should be added there.

Reply:
> This fix is about the wrong assert, but when I fixed it I found out the issue with ItHighAvailablePartitionsRecoveryTest#testHaRecoveryOnZoneTimersRestoreAfterNodeRestart, because assertion starts to work. So, this test fixed here also.

Okay, please indicate this in the ticket description so that it is clearer which test checks this.

If I understand correctly, this ticket should fix the issue from [IGNITE-24057](https://issues.apache.org/jira/browse/IGNITE-24057)? If so, then without a test this assumption may be wrong and it is better to make sure by writing a test.